docs/defradb/guides/merkle-crdt.md

```diff
@@ -2,66 +2,197 @@
+There are three main semantics in CRDTs:
 
-## Delta State Semantics
+| Type           | Description                                 | Efficiency       |
+|----------------|---------------------------------------------|------------------|
+| Operation-based| Sends intent of operation (e.g., “+1”)      | Moderate         |
+| State-based    | Sends full current state                    | Least efficient  |
+| Delta-based    | Sends minimal change (delta)                | Most efficient   |
 
-There are two types of Delta State Semantics: Operation-Based CRDTs and State-Based CRDTs. Operation-Based CRDTs use the intent of an operation as the body or content of the message, while State-Based CRDTs use the resulting state as the body or content of the message. Both have their own advantages and disadvantages, and the appropriate choice depends on the specific use case. Operation-Based CRDTs express actions such as setting a value to 10 or incrementing a counter by 4 through the intent of the operation. State-Based CRDTs, on the other hand, include the resulting state in the message. For example, a message to set a value to 10 would include the value 10 as the body or content of the message.
+**Delta CRDTs** are best suited for resource-constrained systems:
 
-Operation-Based CRDTs tend to be smaller because their messages only contain the operation being performed, while State-Based CRDTs are larger because their messages contain both the current state and the state being changed. It is important to consider the trade-offs between these two types of Delta State Semantics when choosing which one to use in a given situation.
+- Require less bandwidth
+- Lower memory and compute overhead
+- Enable fast and lightweight updates
 
-Delta State Semantics is an optimization of the State-based CRDTs. While both Operation-based CRDTs and State-based CRDTs have their own pros and cons, Delta State CRDTs offer a hybrid approach that uses the state as the message content, but with the same size as an operation.
+**Example**:
 
-In a Delta State CRDT, the message body includes only the minimum amount, or "delta," necessary to transform the previous state to the target state. For example, if we have a set of nine fruit names, and we want to add a banana to the set, the Delta State CRDT would only include the delta, or the value "banana," rather than expressing the entire set of 10 fruit names as in traditional State-based CRDTs. This is like an operation because it has the size of only one action, but it expresses the difference in state between the previous and target rather than the intent of the action.
+Instead of transmitting a whole list, a delta CRDT for a shopping list update may just send: `["+banana"]`.
 
+---
 
 ## Branching and Merging State
 
+### Branching in Merkle CRDTs
+
+Branching occurs when multiple updates happen concurrently from the same base node. Each update becomes a new branch in the DAG.
```

Comment:
todo: Branching occurs when updates happen concurrently from ***different*** nodes. Each ***node*** maintains its own independent branch in the DAG.